db/commitlog/commitlog_replayer.cc

```diff
@@ -199,6 +200,8 @@ db::commitlog_replayer::impl::recover(sstring file, const sstring& fname_prefix)
             f.get();
         } catch (commitlog::segment_data_corruption_error& e) {
             s->corrupt_bytes += e.bytes();
+        } catch (commitlog::segment_truncation& e) {
+            s->truncated_at = e.position();
```

Comment:
nit: `truncated_at` is a loded term, referring to time of truncation.
better call it `truncate_position`.